internal/timeout/timeout.go

```diff
@@ -82,5 +86,35 @@ func Parse(timeout string) (Setting, error) {
 		return Setting{}, fmt.Errorf("unable to parse timeout string %q: %w", timeout, err)
 	}
 
+	if d == 0 {
+		return DefaultSetting(), nil
+	}
+
+	return DurationSetting(d), nil
+}
+
+// ParseMaxAge parses string representations of "max age" values used mostly
+// in cache related settings. An example of this is the MaxAge field of the
+// CORS policy:
+//	- an empty string means "use the default".
+//	- the "disabled" string means "disable cache", so it is turned into a 0 duration.
+//	- a valid Go duration string is used as the specific timeout value.
+//	- any other input means "use the default".
+func ParseMaxAge(timeout string) (Setting, error) {
+	if timeout == "disabled" {
+		return DisabledSetting(), nil
+	}
```

Comment:
I think consensus was we don't to support this magic string, since 0 explicitly specified means "disable cache". Please update the godoc to match as well.

Reply:
I thought that supporting both was preferred 😅 . I'll change it.

Reply:
sorry for the back and forth!

Reply:
done and main rebased.